### PR TITLE
Update readme and related files

### PR DESCRIPTION
**Pull request by oafischer**
Feel free to make changes or suggestions. Is pointing to the Playbook enough? I feel like the external contribution guide covers the most important stuff outside of the Playbook.

### CODE REVIEW

**Review by cadawson — APPROVED** [review]
Looks good. Thanks for updating.

File: README.adoc [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+* https://stolostron.github.io/rhacm-docs/[Internal Documentation]
+
+[#contributing]
+== Contributing
+
+If you are a developer and are interested in contributing or submitting a doc issue, please see the ACM Playbook for the correct procedure. If you are an external user, please see link:EXTERNAL_CONTRIBUTING.adoc#red-hat-advanced-cluster-management-for-kubernetes-contributing-external[Contributing to Red Hat Advanced Cluster Management for Kubernetes staged documentation].
+
+[#owners]
+== Owners
```
Comment (swopebe) [inline README.adoc]: we collectively decided we didn't need or want to put owners here.

File: EXTERNAL_CONTRIBUTING.adoc [hunk 2]
```diff
@@ -1,40 +1,38 @@
...
 
-The documentation source is in ASCIIDoc format, and the content that is provided in PRs must conform to the ASCIIDoc requirements. You can find some guidance for authoring in ASCIIDoc in the https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/[AsciiDoc Syntax quick reference].
+The documentation source is in https://asciidoc.org/[AsciiDoc] format, and the content that is provided in PRs must conform to the AsciiDoc requirements. You can find some guidance for authoring in AsciiDoc in the https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/[AsciiDoc Syntax Quick Reference].
 
 Note: All PRs outside of a typo fix require an issue.
 
 Open an issue here: https://github.com/stolostron/rhacm-docs/issues/new[https://github.com/open-cluster-management/rhacm-docs/issues/new]. Complete the template with the information that is required for your requested change. 
 
-[#How-is-my-issue-handled]
+[#how-is-my-issue-handled]
```
Comment (swopebe) [inline EXTERNAL_CONTRIBUTING.adoc:21]: this will change with Jira, but fine to keep for now.

**Review by swopebe — APPROVED** [review]
Looks great! Couple comments for consideration.

File: README.adoc (hunk 1 above)
Comment (oafischer) [reply README.adoc]: Removed it, thanks for letting me know!

**Pull request merged**